src/ManifoldMuseum.jl

```diff
@@ -87,6 +92,35 @@ then implements the feature itself.
 """
 @traitdef IsDecoratorManifold{M}
 
+"""
+    manifold(M::Manifold; recurse::Bool=false)
+
+If `M` is a decorator manifold, return the underlying manifold. if `recurse` is
+`true`, continue un-decorating until a non-decorator manifold is encountered.
+"""
+function manifold end
+
+@traitfn function manifold(M::MT; recurse::Bool=false) where {MT<:Manifold;IsDecoratorManifold{MT}}
```

Comment:
Are you sure the `recurse` keyword argument is actually necessary? I think normal dispatch should handle multiple layers of unwrapping well.

Reply:
By the way, I think the name of this function is not specific enough, maybe something like `unwrap_manifold` would be better?

Reply:
Or maybe `basic_manifold` in the sense of an undecorated one?

Reply:
> Are you sure the recurse keyword argument is actually necessary?

Most of the time we only want to unwrap one layer of a multiply-decorated manifold. But sometimes we may want to get the underlying topological manifold by unwrapping all layers (e.g. `manifold_dimension`). Perhaps 2 different functions make sense: `base_manifold` for unwrapping one layer and `topological_manifold` for unwrapping all layers.

Reply:
but you never know the order of the decorators, so there is not much use in saying “one” layer. You might have a `DecoA{DecoB{M}}` or a `DecoB{DecoA{M}}` – so you're either looking for the manifold (itself) and want `M` or you are looking for a specific decorator (since you want to access the metric directly for example). `base_manifold` would return `M`. And maybe to access properties we better do `get`ers?

Reply:
I think you're right, but it will be easier to tell when we get to multiply-decorated manifolds, like Lie groups equipped with metrics, which should have all of the functionality of the Lie group and of a metric manifold. And I think in a case like that, a specific ordering of decorators makes more sense. (ie I think it makes more sense to equip the Lie group with a metric than to equip a metric manifold with a group operation).

Reply:
Multiple decorators is an important issue. One day, probably not that far away, we will start asking questions like "what metric should have a quotient manifold of a riemannian manifold with a custom metric by an action of a Lie group of rotations with a custom metric?" and even figuring out the correct order of decorators will be a problem.

Is there actually a reason to allow more than one (likely partial) order of decorators?

Reply:
Of course there might be non-useful orderings – but should we be so strict to employ that if we can (in my opinion quite easily) ignore ordering by saying (with the trait) just say: Throw away deco until you find the on I need for my metric.

In short, my counter question is: Are we restricting too much by imposing a (eventually partial) order? I am just not sure ;)

Reply:
It's hard to decide at this moment, there may be issues that are order-dependent. Something like "unwrap decorators to get X" functions should be good enough for now.